### PR TITLE
[Berkeley] Remove evaluated columns from linearization

### PR DESCRIPTION
**Pull request by mrmr1993**
This PR builds upon https://github.com/MinaProtocol/mina/pull/13529, removing the 'non-optional' gate selectors from the linearization argument in pickles, as part of the work to remove the linearization from kimchi proofs entirely. The proof-systems counterpart can be found in https://github.com/o1-labs/proof-systems/pull/1106.

Most notably, this fixes an issue where one of the scalars for one of the gates was underconstrained, potentially allowing pickles to 'verify' a malicious invalid proof.

Checklist:

- [ ] Dependency versions are unchanged
  - Notify Velocity team if dependencies must change in CI
- [ ] Modified the current draft of release notes with details on what is completed or incomplete within this project
- [ ] Document code purpose, how to use it
  - Mention expected invariants, implicit constraints
- [ ] Tests were added for the new behavior
  - Document test purpose, significance of failures
  - Test names should reflect their purpose
- [ ] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them

### CODE REVIEW

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Review by rbonichon — APPROVED** [review]
LGTM

**Comment by dannywillems** [comment]
Comments in composition_types are outdated. L119.

File: src/lib/pickles/pickles.ml [hunk 1]
```diff
@@ -652,7 +652,7 @@ module Make_str (_ : Wire_types.Concrete) = struct
         let tag, _, p, Provers.[ step ] =
           Common.time "compile" (fun () ->
               compile_promise () ~public_input:(Input Field.typ)
-                ~auxiliary_typ:Typ.unit
+                ~override_wrap_domain:N1 ~auxiliary_typ:Typ.unit
```
[the remark this answers is not in the document]
Comment (rbonichon) [reply src/lib/pickles/pickles.ml:655]: Disregard the previous comment. Should we prefix with `Proof_verified`?

[the remark this answers is not in the document]
Comment (dannywillems) [reply src/lib/pickles/pickles.ml:655]: Is it not something left from testing?

[the remark this answers is not in the document]
Comment (mrmr1993) [reply src/lib/pickles/pickles.ml]: No, it's important.

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by rbonichon** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by dannywillems** [comment]
!ci-build-me

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**